common/route/router2.cc

```diff
@@ -1180,6 +1180,37 @@ struct Router2
         }
     }
 
+    void write_utilisation_by_wiretype_heatmap(std::ostream &out)
+    {
+        dict<IdString, int> util_by_type;
+        for (auto &wd : flat_wires) {
+            IdString type = ctx->getWireType(wd.w);
+            if (wd.curr_cong > 0) {
+                if (util_by_type.count(type))
+                    util_by_type[type] += 1;
```

Comment:
I think this should be increasing by `curr_cong` not by a fixed 1, because if two nets are using a wire, you want it to have a utilisation of 2, too

Reply:
Maybe there should be both. If you increase by `curr_cong`, then you can imagine a situation early in routing where 50% of the wires have `curr_cong == 2` and 50% of the wires have `curr_cong == 0`; it would report 100% utilisation, when there's still plenty of space to spread out to. I think that would be a different situation to "actually *all* of the wires here are used and the router is struggling", and a fixed-1 count would let one distinguish between those situations.